### ISSUE TITLE
@lang full-text search doesn't work as documented.

### ISSUE THREAD

**Issue by danielmai** [dgraph-io/dgraph#3111]
- What version of Dgraph are you using?
v1.0.12

- Have you tried reproducing the issue with latest release?
Yes. I also tried in previous releases (v1.0.7 and above).

- What is the hardware spec (RAM, OS)?
n/a

- Steps to reproduce the issue (command/config used to run Dgraph).

Follow schema/query/mutation from "Predicates i18n" example (https://docs.dgraph.io/query-language/#predicates-i18n):

Schema:
```graphql
<公司>: string @index(fulltext) @lang .
```
Mutation:
```graphql
{
  set {
    _:a <公司> "Dgraph Labs Inc"@en .
    _:b <公司> "夏新科技有限责任公司"@zh .
  }
}
```
Query:
```graphql
query {
  q(func: alloftext(<公司>@., <夏新科技有限责任公司>)) {
    _predicate_
  }
}
# Response:
# {
#   "data": {
#     "q": []
#   }
# }
```

- Expected behaviour and actual result.

The last query does not return the node `_:b <公司> "夏新科技有限责任公司"@zh .`, even though I think `alloftext(pred@., ...)` would search for all untagged preds and then any language pred.

When I change `<公司>@.` to `<公司>@zh` in the alloftext func it works as expected.
```graphql
query {
  q(func: alloftext(<公司>@zh, <夏新科技有限责任公司>)) {
    <公司>@.
    _predicate_
  }
}
# Response:
# {
#   "data": {
#     "q": [
#       {
#         "公司@.": "夏新科技有限责任公司",
#         "_predicate_": [
#           "公司"
#         ]
#       }
#     ]
#   }
# }
```

What confused me is when this worked: I get a result when searching `<公司>@.` for `<dgraph>`.

```graphql
query {
  q(func: alloftext(<公司>@., <dgraph>)) {
    <公司>@.
    _predicate_
  }
}
# Response:
# {
#   "data": {
#     "q": [
#       {
#         "公司@.": "Dgraph Labs Inc",
#         "_predicate_": [
#           "公司"
#         ]
#       }
#     ]
#   }
# }
```

**Comment by danielmai** [comment]
@srfrog should the documented example work?

**Comment by srfrog** [comment]
The example was working, that's why I put it there. This seems like a regression. Thanks I'll investigate.

**Comment by srfrog** [comment]
Quick test, this works:
```
query {
  q(func: alloftext(<公司>@., "Dgraph Labs Inc")) {
    _predicate_
  }
}
...
	"data": {
		"q": [
			{
				"_predicate_": [
					"公司"
				]
			}
		]
	}
```
So clearly something has changed.

**Comment by danielmai** [comment]
Yeah, that's what I ran too (shown at the end of the issue).

**Comment by srfrog** [comment]
Sorry I take it back, it's not a regression it's correct behavior. In the case of **allof** and **anyof** functions we need to know the specific language index to generate the correct search tokens. The reason that `@.` works with English value is that `@.` is converted to `@en`. For `@.` to work as we hope here, we would need to know all the language indexes for the predicate and try all of them for the same value. In theory we could do it, but it's not a feature now.

In short, the query example is wrong. It needs to be `@zh` instead of `@.`.

**Comment by danielmai** [comment]
Cool. Will update the docs accordingly. Thanks @srfrog.
